### PR TITLE
Simplify progress updates to one call per control step

### PR DESCRIPTION
**Pull request by cvolkcvolk**
Simplify `ProgressTracker` to advance once per control step through `TaskSuccessTerm`. Remove duplicate-step filtering and skipped-step recovery while keeping predicate-result reuse within each update and per-environment resets.

## Why this change?

Stacked on #1304. Isaac Lab already calls `TaskSuccessTerm` once per control step, and `ProgressTrackingRecorder` only reads the result. The extra step-index bookkeeping supports repeated or skipped tracker updates that the normal environment loop does not perform.

## What changes?

- Remove the stored last-step indices, update masks for repeated calls, and skipped-step counter resets.
- Keep `step_index` as optional event metadata. It no longer controls whether progress advances.
- Keep active-predicate masks and the shared per-update cache, so progress advancement and final-condition checks do not count the same requirement twice.
- Update tests and documentation to use one update per control step and passive reads afterward.

Task declarations do not change. Direct callers must call `ProgressTracker.step()` exactly once per control step; repeated calls are no longer ignored. Manually invoking `TaskSuccessTerm` is not a passive success check. The outdated cable-routing-v2 demo's direct success-term call remains outside this change and needs migration with its existing API usage.

## Usage example

Before, repeating the same index was ignored:

```python
tracker.step(env, step_index=env.episode_length_buf)
tracker.step(env, step_index=env.episode_length_buf)
success = tracker.is_complete()
```

After, update once and read the stored result as needed:

```python
tracker.step(env, step_index=env.episode_length_buf)
success = tracker.is_complete()
progress = tracker.get_state()
```

## Validation

- All pre-commit hooks passed with `pre-commit run --all-files`.
- 50 simulation-independent test bodies passed in Docker, covering temporal requirements, objective tracking, composition, and success-term updates/resets. These were called directly, without the simulator wrapper.
- Normal focused pytest was blocked before the test body by duplicate `--enable_cameras` registration, followed by missing `omni.timeline` during teardown. The same failure reproduces on unchanged #1304.
- Five broader task/builder checks could not import the locally missing `isaaclab_tasks.contrib` dependency. The unrelated CAP cable-task check was excluded. Full simulator validation remains outstanding.

### CODE REVIEW

File: isaaclab_arena/progress_tracking/progress_tracker.py [hunk 1]
```diff
@@ -476,32 +467,18 @@ def _all_objectives_complete(runners: list[ProgressObjectiveRunner]) -> torch.Te
...
-            ), "Reset progress before restarting step indices."
-            updated_envs = step_index > self._last_processed_step
-            if not bool(updated_envs.any()):
-                return
-            # Unobserved control steps cannot contribute to a consecutive streak.
-            skipped_steps = (self._last_processed_step >= 0) & (step_index > self._last_processed_step + 1)
-            for runner in self.runners:
-                runner._reset_consecutive_step_requirements(skipped_steps)
-        active_envs = updated_envs.clone()
+        active_envs = torch.ones(self.num_envs, dtype=torch.bool, device=self.device)
```
Comment (xyao-nv) [inline isaaclab_arena/progress_tracking/progress_tracker.py]: Is there any specific reason why the special handling of step = 1, 1, 2 is removed?
`skipped_steps = (self._last_processed_step >= 0) & (step_index > self._last_processed_step + 1)`

File: isaaclab_arena/progress_tracking/progress_tracker.py [hunk 2]
```diff
@@ -476,32 +467,18 @@ def _all_objectives_complete(runners: list[ProgressObjectiveRunner]) -> torch.Te
...
-            step_index is not None or not self._requires_step_index
-        ), "TrueForConsecutiveStepsCfg requires a per-environment step_index."
-        updated_envs = torch.ones(self.num_envs, dtype=torch.bool, device=self.device)
         if step_index is not None:
             assert step_index.shape == (self.num_envs,), "step_index must contain one index per environment."
             assert step_index.dtype in (torch.int32, torch.int64), "step_index must contain integer indices."
-            step_index = step_index.to(device=self.device)
             assert bool((step_index >= 0).all()), "step_index must be non-negative."
-            assert bool(
-                (step_index >= self._last_processed_step).all()
```
Comment (xyao-nv) [inline isaaclab_arena/progress_tracking/progress_tracker.py]: How about this index jumping case? step =1, 3, 4

See full discussion on slack.
